### PR TITLE
Add url_for_version function and gtk variant

### PR DESCRIPTION
**Pull request by adamjstewart**
I messed up #427 when trying to resolve conflicts, so here it is again.

### CODE REVIEW

File: var/spack/repos/builtin/packages/qt/package.py [hunk 1]
```diff
@@ -1,41 +1,31 @@
...
+    variant('gtk',        default=False, description="Build with gtkplus.")
+
     patch('qt3krell.patch', when='@3.3.8b+krellpatch')
 
     # Use system openssl for security.
     #depends_on("openssl")
 
     depends_on("glib")
-    depends_on("gtkplus")
+    depends_on("gtkplus", when='+gtk')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/qt/package.py]: @mathstuf: @adamjstewart: @lee218llnl  do you know of anything that relies on GTK support in Qt?  This change makes `gtk` off by default, so other packages may need to ask specifically for it.  Is it better to build Qt with gtk by default?

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qt/package.py]: This was done by request of #166. I'll let you guys decide whether it should default to True or False.

**Comment by tgamblin** [comment]
Asked @alfredo-gimenez.  I do not believe he needs this for MemAxes.  I think that is all.  If there are packages that need Qt/GTK they will need to update their variants.  Seems reasonable to have this off by default.

**Comment by tgamblin** [comment]
@alfredo-gimenez points out that when `~gtk`, there should probably be a config option to disable `gtk`, or this will pull in a system `gtk`.

**Comment by adamjstewart** [comment]
I'll add another PR to fix that. I don't think I can just push to the same branch, right?

**Comment by mathstuf** [comment]
You can, but will need a new PR with a differently named branch or not.

**Pull request merged**